src/hooks/useMediaQuery/index.js

```diff
@@ -0,0 +1,31 @@
+import { useLayoutEffect, useState, useMemo } from "react";
+
+export function useMediaQuery(query) {
+  const queries = useMemo(() => {
+    return Array.isArray(query) ? query : [query];
+  }, [query]);
+
+  const [matches, setMatches] = useState(queries.map(query => !!window.matchMedia(query).matches));
+
+  useLayoutEffect(() => {
+    const mediaQueryList = queries.map(query => window.matchMedia(query));
+    const updated = mediaQueryList.map((query, index) => {
+      const callback = event => {
```

Comment:
maybe add a small comment about this callback ? that we would like to listen to changes on the window and update this state id this query match / not match after each update?